pkg/cmd/extension/command.go

```diff
@@ -39,6 +45,148 @@ func NewCmdExtension(f *cmdutil.Factory) *cobra.Command {
 	}
 
 	extCmd.AddCommand(
+		func() *cobra.Command {
+			query := search.Query{
+				Kind: search.KindRepositories,
+			}
+			qualifiers := search.Qualifiers{
+				Topic: []string{"gh-extension"},
+			}
+			var order string
+			var sort string
+			var webMode bool
+			var exporter cmdutil.Exporter
+
+			cmd := &cobra.Command{
+				Use:   "search [<query>]",
+				Short: "Search available gh extensions",
+				Long: heredoc.Doc(`
+				Search for gh extensions.
+
+				This command behaves similarly to 'gh search repos' but does not
+				support as many search qualifiers. For a finer grained search of
+				extensions, try using:
+
+				gh search repos --topic "gh-extension"
+
+				and adding qualifiers as needed. See 'gh help search repos' to learn
+				more about repository search.
+`),
+				RunE: func(cmd *cobra.Command, args []string) error {
+					cfg, err := config()
+					if err != nil {
+						return err
+					}
+					client, err := httpClient()
+					if err != nil {
+						return err
+					}
+
+					if cmd.Flags().Changed("order") {
+						query.Order = order
+					}
+					if cmd.Flags().Changed("sort") {
+						query.Sort = sort
+					}
+
+					query.Keywords = args
+					query.Qualifiers = qualifiers
+
+					host, _ := cfg.DefaultHost()
+					searcher := search.NewSearcher(client, host)
+					tp := tableprinter.New(io)
+
+					if webMode {
+						url := searcher.URL(query)
+						if io.IsStdoutTTY() {
+							fmt.Fprintf(io.ErrOut, "Opening %s in your browser.\n", text.DisplayURL(url))
+						}
+						return browser.Browse(url)
+					}
+
+					result, err := searcher.Repositories(query)
+					if err != nil {
+						return err
+					}
+
+					if io.CanPrompt() {
+						fmt.Fprintf(io.Out, "Showing %d of %d gh extensions\n", query.Limit, result.Total)
+						fmt.Fprintln(io.Out)
+						tp.HeaderRow("NAME", "REPO", "INSTALLED", "OFFICIAL", "DESCRIPTION")
```

Comment:
Nit: HeaderRow automatically prints only in TTY mode and not in other modes, so it doesn't have to be wrapped in any conditional